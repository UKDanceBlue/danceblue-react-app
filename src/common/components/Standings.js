import React, { useEffect, useState } from 'react';
import { Text, View, StyleSheet, TouchableOpacity, ActivityIndicator } from 'react-native';
import Place from './Place';
import { globalColors, globalStyles, globalTextStyles } from '../../theme';

/**
 * Standings implementation for the a generic leaderboard
 * @param {function} standingData an array of the information to be displayed (defalt: blank)
 * @param {bool} expandable Can the standings be expanded/contracted (defalt: false)
 * @param {bool} expanded Are the standings expanded by default (defalt: false)
 * @param {number} collapsedRows How many rows should be shown when collapsed (default: 3)
 */
const Standings = ({ standingData, expandable, startExpanded, collapsedRows = 3 }) => {
  const [rows, setRows] = useState([]);
  const [expanded, setExpanded] = useState(!!startExpanded);
  const [rowsToShow, setRowsToShow] = useState(collapsedRows);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(
    () => setRowsToShow(expanded ? standingData.length : collapsedRows),
    [expanded, standingData, collapsedRows]
  );

  useEffect(() => {
    setIsLoading(true);
    const sortedStandings = standingData;
    sortedStandings.sort((a, b) => b.points - a.points);
    const tempRows = [];
    for (let i = 0; i < sortedStandings.length && i < rowsToShow; i++) {
      tempRows.push(
        <Place
          key={sortedStandings[i].id}
          rank={i + 1}
          name={sortedStandings[i].name}
          points={sortedStandings[i].points}
          isHighlighted={sortedStandings[i].highlighted}
        />
      );
      setRows(tempRows);
    }
    setIsLoading(false);
  }, [standingData, rowsToShow]);

  return (
    <View style={globalStyles.genericView}>
      <View style={localStyles.ListView}>
        {!isLoading && (
          <>
            {rows}
            {expandable &&
              (expanded ? (
                <TouchableOpacity onPress={() => setExpanded(false)}>
                  <Text style={globalTextStyles.italicText}>Show less...</Text>
                </TouchableOpacity>
              ) : (
                <TouchableOpacity onPress={() => setExpanded(true)}>
                  <Text style={globalTextStyles.italicText}>Show more...</Text>
                </TouchableOpacity>
              ))}
          </>
        )}
      </View>
      {isLoading && (
        <ActivityIndicator
          size="large"
          color={globalColors.lightBlue}
          style={{
            alignItems: 'center',
            justifyContent: 'center',
            padding: 20,
          }}
        />
      )}
    </View>
  );
};

const localStyles = StyleSheet.create({
  ListView: {
    paddingLeft: 5,
    paddingTop: 5,
    paddingBottom: 5,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    backgroundColor: 'white',
    flex: 1,
  },
});

export default Standings;
