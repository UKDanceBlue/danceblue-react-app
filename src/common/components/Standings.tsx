/// <reference types="react" />
import React, { ReactElement, useEffect, useState } from 'react';
import { View, StyleSheet, TouchableOpacity, ActivityIndicator } from 'react-native';
import { Text } from 'react-native-elements';
import Place from './Place';
import { globalColors, globalStyles, globalTextStyles } from '../../theme';
import { StandingType } from '../../types/StandingType';

/**
 * Standings implementation for the a generic leaderboard
 */
const Standings = ({
  titleText,
  standingData,
  expandable = false,
  startExpanded = false,
  collapsedRows = 3,
  lastRow,
  dadJokeTempMagic = false,
  dadJokeTempMagicCallback = () => {},
}: {
  titleText: string;
  standingData: StandingType[];
  expandable?: boolean;
  startExpanded?: boolean;
  showTrophies?: boolean;
  collapsedRows?: number;
  dadJokeTempMagic?: boolean;
  dadJokeTempMagicCallback?: (arg0: boolean, arg1: string) => unknown;
}) => {
  const [rows, setRows] = useState<ReactElement[]>([]);
  const [expanded, setExpanded] = useState<boolean>(!!startExpanded);
  const [rowsToShow, setRowsToShow] = useState<number>(collapsedRows);
  const [isLoading, setIsLoading] = useState<boolean>(true);

  useEffect(
    () => setRowsToShow(expanded ? standingData.length : collapsedRows),
    [expanded, standingData, collapsedRows]
  );

  useEffect(() => {
    setIsLoading(true);
    const sortedStandings = standingData.map((standing) => ({
      ...standing,
      points: standing.points || 0,
    }));
    sortedStandings.sort((a, b) => b.points - a.points);
    const tempRows = [];
    for (let i = 0; i < sortedStandings.length && i < rowsToShow; i++) {
      tempRows.push(
        <Place
          key={sortedStandings[i].id}
          rank={i + 1}
          name={sortedStandings[i].name}
          points={sortedStandings[i].points}
          isHighlighted={sortedStandings[i].highlighted}
          lastRow={i === sortedStandings.length - 1 && i === rowsToShow - 1}
          dadJokeTempMagic={dadJokeTempMagic}
          dadJokeTempMagicCallback={(arg0: boolean) =>
            dadJokeTempMagicCallback(arg0, sortedStandings[i].id)
          }
        />
      );
    }
    setRows(tempRows);
    setIsLoading(false);
  }, [standingData, rowsToShow]);

  return (
    <View style={globalStyles.genericView}>
      <View style={localStyles.ListView}>
        {!isLoading && (
          <>
            {rows}
            {expandable &&
              (expanded ? (
                <TouchableOpacity onPress={() => setExpanded(false)}>
                  <Text style={globalTextStyles.italicText}>Show less...</Text>
                </TouchableOpacity>
              ) : (
                <TouchableOpacity onPress={() => setExpanded(true)}>
                  <Text style={globalTextStyles.italicText}>Show more...</Text>
                </TouchableOpacity>
              ))}
          </>
        )}
      </View>
      {isLoading && (
        <ActivityIndicator
          size="large"
          color={globalColors.lightBlue}
          style={{
            alignItems: 'center',
            justifyContent: 'center',
            padding: 20,
          }}
        />
      )}
    </View>
  );
};

const localStyles = StyleSheet.create({
  ListView: {
    paddingLeft: 5,
    paddingTop: 5,
    paddingBottom: 5,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    backgroundColor: 'white',
    flex: 1,
  },
});

export default Standings;
