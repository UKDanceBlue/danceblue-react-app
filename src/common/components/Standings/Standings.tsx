import { Text } from "native-base";
import { ReactElement, useEffect, useState } from "react";
import { ActivityIndicator, StyleSheet, TouchableOpacity, View } from "react-native";

import { globalColors, globalStyles, globalTextStyles } from "../../../theme";
import { StandingType } from "../../../types/StandingType";
import Place from "../Place";

/**
 * Standings implementation for the a generic leaderboard
 */
const Standings = ({
  standingData,
  expandable = false,
  startExpanded = false,
  collapsedRows = 3,
}: {
  titleText: string;
  standingData: StandingType[];
  expandable?: boolean;
  startExpanded?: boolean;
  showTrophies?: boolean;
  collapsedRows?: number;
  dadJokeTempMagicCallback?: (arg0: boolean, arg1: string) => unknown;
}) => {
  const [ rows, setRows ] = useState<ReactElement[]>([]);
  const [ expanded, setExpanded ] = useState<boolean>(!!startExpanded);
  const [ rowsToShow, setRowsToShow ] = useState<number>(collapsedRows);
  const [ isLoading, setIsLoading ] = useState<boolean>(true);

  useEffect(
    () => setRowsToShow(expanded ? standingData.length : collapsedRows),
    [
      expanded, standingData, collapsedRows
    ]
  );

  useEffect(() => {
    setIsLoading(true);
    const sortedStandings = standingData.map((standing) => ({
      ...standing,
      points: standing.points || 0,
    }));
    sortedStandings.sort((a, b) => {
      if (b.points === a.points) {
        if (a.name > b.name) {
          return 1;
        } else {
          return 0;
        }
      } else {
        return b.points - a.points;
      }
    });
    const tempRows = [];
    for (let i = 0; i < sortedStandings.length && i < rowsToShow; i++) {
      tempRows.push(
        <Place
          key={sortedStandings[i].id}
          rank={i + 1}
          name={sortedStandings[i].name}
          points={sortedStandings[i].points}
          isHighlighted={sortedStandings[i].highlighted}
          lastRow={i === sortedStandings.length - 1 && i === rowsToShow - 1}
        />
      );
    }
    setRows(tempRows);
    setIsLoading(false);
  }, [ standingData, rowsToShow ]);

  return (
    <View style={globalStyles.genericView}>
      <View style={localStyles.ListView}>
        {!isLoading && (
          <>
            {rows}
            {expandable &&
              (expanded ? (
                <TouchableOpacity onPress={() => setExpanded(false)}>
                  <Text style={globalTextStyles.italicText}>Show less...</Text>
                </TouchableOpacity>
              ) : (
                <TouchableOpacity onPress={() => setExpanded(true)}>
                  <Text style={globalTextStyles.italicText}>Show more...</Text>
                </TouchableOpacity>
              ))}
          </>
        )}
      </View>
      {isLoading && (
        <ActivityIndicator
          size="large"
          color={globalColors.lightBlue}
          style={{
            alignItems: "center",
            justifyContent: "center",
            padding: 20,
          }}
        />
      )}
    </View>
  );
};

const localStyles = StyleSheet.create({
  ListView: {
    alignItems: "flex-start",
    backgroundColor: "white",
    flex: 1,
    justifyContent: "flex-start",
    paddingBottom: 5,
    paddingLeft: 5,
    paddingTop: 5,
  },
});

export default Standings;
