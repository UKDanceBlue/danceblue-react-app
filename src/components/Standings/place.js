import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome5'

/**
 * TODO
 * @param {Object} props Properties of the component: (TODO)
 * @returns A React Native component
 * @author Kenton Carrier
 * @since 1.0.1
 * @class
 */
class Place extends React.Component {
  /**
   * Called to generate a React Native component
   * @returns A JSX formatted component
   */
  render () {
    // The 'top3Icon function adds an award icon to the top 3 teams
    const top3Icon = rank => {
      if (rank === 1) {
        return <Icon name='award' size={30} color='gold' />
      } else if (rank === 2) {
        return <Icon name='award' size={30} color='silver' />
      } else if (rank === 3) {
        return <Icon name='award' size={30} color='blue' />
      }
    }
    return (
      // Renders the individual row of the leaderboard for each team
      <View style={
        this.props.isHighlighted
          ? styles.highlightedRow
          : styles.row
        }
      >
        <View style={styles.left}>
          <Text style={styles.place}>
            {/* Renders the rank/place of the team */}
            {this.props.rank}
          </Text>
          {/* Calls the top3Icon function */}
          {top3Icon(this.props.rank)}
        </View>
        <View style={styles.middle}>
          <Text style={styles.name}>
            {/* Renders the team name */}
            {this.props.teamName}
          </Text>
          <Text style={styles.team}>
            {/* Renders the team number */}
            {this.props.teamNumber}
          </Text>
        </View>
        <View style={styles.right}>
          <Text style={styles.bold}>
            {/* Renders the number of points earned */}
            {this.props.showPointsPerMember
              ? this.props.pointsPerMember < 1
                  ? '< 1 '
                  : this.props.pointsPerMember
              : this.props.points}
            {this.props.showPointsPerMember
              ? ' points'
              : this.props.points === 1
                ? ' point'
                : ' points'}
          </Text>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  row: {
    paddingTop: 10,
    paddingBottom: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderBottomWidth: 0.2,
    borderBottomColor: '#999999'
  },
  highlightedRow: {
    paddingTop: 10,
    paddingBottom: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderBottomWidth: 0.2,
    borderBottomColor: '#999999',
    backgroundColor: '#ffffc8'
  },
  left: {
    width: '20%',
    flexDirection: 'row'
  },
  right: {
    width: '30%',
    justifyContent: 'flex-end'
  },
  middle: {
    width: '50%',
    flexDirection: 'column'
  },
  place: {
    fontSize: 20,
    fontWeight: 'bold',
    marginLeft: 10,
    marginRight: 10
  },
  bold: {
    fontWeight: 'bold',
    textAlign: 'right',
    paddingRight: 10,
    fontSize: 17
  },
  name: {
    fontWeight: 'bold',
    fontSize: 15
  },
  team: {
    fontWeight: 'normal',
    fontStyle: 'italic'
  }
})

export default Place
