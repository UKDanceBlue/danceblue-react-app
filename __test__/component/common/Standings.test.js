import React from 'react';
import renderer from 'react-test-renderer';
import { loadAsync } from 'expo-font';
import { FontAwesome5, MaterialIcons } from '@expo/vector-icons';
import Standings from '../../../src/common/components/Standings';

beforeAll(async () => loadAsync(FontAwesome5.font));
beforeAll(async () => loadAsync(MaterialIcons.font));

describe('Standings component', () => {
  jest.useFakeTimers();
  test('renders correctly with no highlighted paramter', () => {
    const standingData = [
      {
        id: '1',
        name: 'Test 1',
        points: 111,
      },
      {
        id: '2',
        name: 'Test 2',
        points: 222,
      },
      {
        id: '3',
        name: 'Test 3',
        points: 333,
      },
      {
        id: '4',
        name: 'Test 4',
        points: 444,
      },
      {
        id: '5',
        name: 'Test 5',
        points: 555,
      },
    ];
    const tree = renderer.create(<Standings standingData={standingData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly expanded', () => {
    const standingData = [
      {
        highlighted: false,
        id: '1',
        name: 'Test 1',
        points: 111,
      },
      {
        highlighted: false,
        id: '2',
        name: 'Test 2',
        points: 222,
      },
      {
        highlighted: false,
        id: '3',
        name: 'Test 3',
        points: 333,
      },
      {
        highlighted: false,
        id: '4',
        name: 'Test 4',
        points: 444,
      },
      {
        highlighted: false,
        id: '5',
        name: 'Test 5',
        points: 555,
      },
    ];
    const tree = renderer
      .create(<Standings standingData={standingData} expandable startExpanded />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with 4 collapsed rows', () => {
    const standingData = [
      {
        highlighted: false,
        id: '1',
        name: 'Test 1',
        points: 111,
      },
      {
        highlighted: false,
        id: '2',
        name: 'Test 2',
        points: 222,
      },
      {
        highlighted: false,
        id: '3',
        name: 'Test 3',
        points: 333,
      },
      {
        highlighted: false,
        id: '4',
        name: 'Test 4',
        points: 444,
      },
      {
        highlighted: false,
        id: '5',
        name: 'Test 5',
        points: 555,
      },
    ];
    const tree = renderer
      .create(<Standings standingData={standingData} collapsedRows={4} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with a title', () => {
    const standingData = [
      {
        highlighted: false,
        id: '1',
        name: 'Test 1',
        points: 111,
      },
      {
        highlighted: false,
        id: '2',
        name: 'Test 2',
        points: 222,
      },
      {
        highlighted: false,
        id: '3',
        name: 'Test 3',
        points: 333,
      },
      {
        highlighted: false,
        id: '4',
        name: 'Test 4',
        points: 444,
      },
      {
        highlighted: false,
        id: '5',
        name: 'Test 5',
        points: 555,
      },
    ];
    const tree = renderer
      .create(<Standings titleText="Test Title" standingData={standingData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
