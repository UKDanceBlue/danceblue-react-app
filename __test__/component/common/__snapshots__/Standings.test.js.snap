// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Standings component renders correctly expanded 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "flex-start",
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingTop": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F2236",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderTopColor": "#1F2236",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": "20%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 20,
                "marginLeft": 10,
                "marginRight": 10,
              },
            ]
          }
        >
          1
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "gold",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "700",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "textAlign": "center",
            "width": "50%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 15,
              },
            ]
          }
        >
          Test 5
        </Text>
      </View>
      <View
        style={
          Object {
            "justifyContent": "flex-end",
            "textAlign": "right",
            "width": "30%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 17,
                "paddingRight": 10,
              },
            ]
          }
        >
          555
           points
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderTopColor": "#1F2236",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": "20%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 20,
                "marginLeft": 10,
                "marginRight": 10,
              },
            ]
          }
        >
          2
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "silver",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "700",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "textAlign": "center",
            "width": "50%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 15,
              },
            ]
          }
        >
          Test 4
        </Text>
      </View>
      <View
        style={
          Object {
            "justifyContent": "flex-end",
            "textAlign": "right",
            "width": "30%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 17,
                "paddingRight": 10,
              },
            ]
          }
        >
          444
           points
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderTopColor": "#1F2236",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": "20%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 20,
                "marginLeft": 10,
                "marginRight": 10,
              },
            ]
          }
        >
          3
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "blue",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "700",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "textAlign": "center",
            "width": "50%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 15,
              },
            ]
          }
        >
          Test 3
        </Text>
      </View>
      <View
        style={
          Object {
            "justifyContent": "flex-end",
            "textAlign": "right",
            "width": "30%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 17,
                "paddingRight": 10,
              },
            ]
          }
        >
          333
           points
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F2236",
            "fontSize": 15,
            "fontStyle": "italic",
          }
        }
      >
        Show less...
      </Text>
    </View>
  </View>
</View>
`;

exports[`Standings component renders correctly with 4 collapsed rows 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "flex-start",
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingTop": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F2236",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "borderTopColor": "#1F2236",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": "20%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 20,
                "marginLeft": 10,
                "marginRight": 10,
              },
            ]
          }
        >
          1
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "gold",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "700",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "textAlign": "center",
            "width": "50%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 15,
              },
            ]
          }
        >
          Test 5
        </Text>
      </View>
      <View
        style={
          Object {
            "justifyContent": "flex-end",
            "textAlign": "right",
            "width": "30%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 17,
                "paddingRight": 10,
              },
            ]
          }
        >
          555
           points
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderTopColor": "#1F2236",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": "20%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 20,
                "marginLeft": 10,
                "marginRight": 10,
              },
            ]
          }
        >
          2
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "silver",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "700",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "textAlign": "center",
            "width": "50%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 15,
              },
            ]
          }
        >
          Test 4
        </Text>
      </View>
      <View
        style={
          Object {
            "justifyContent": "flex-end",
            "textAlign": "right",
            "width": "30%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 17,
                "paddingRight": 10,
              },
            ]
          }
        >
          444
           points
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderTopColor": "#1F2236",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": "20%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 20,
                "marginLeft": 10,
                "marginRight": 10,
              },
            ]
          }
        >
          3
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "blue",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "700",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "textAlign": "center",
            "width": "50%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 15,
              },
            ]
          }
        >
          Test 3
        </Text>
      </View>
      <View
        style={
          Object {
            "justifyContent": "flex-end",
            "textAlign": "right",
            "width": "30%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 17,
                "paddingRight": 10,
              },
            ]
          }
        >
          333
           points
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderTopColor": "#1F2236",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "width": "20%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 20,
                "marginLeft": 10,
                "marginRight": 10,
              },
            ]
          }
        >
          4
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "textAlign": "center",
            "width": "50%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 15,
              },
            ]
          }
        >
          Test 2
        </Text>
      </View>
      <View
        style={
          Object {
            "justifyContent": "flex-end",
            "textAlign": "right",
            "width": "30%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1F2236",
                "fontSize": 15,
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 17,
                "paddingRight": 10,
              },
            ]
          }
        >
          222
           points
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Standings component renders correctly with a title 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "flex-start",
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingTop": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F2236",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Test Title
    </Text>
  </View>
  <ActivityIndicator
    animating={true}
    color="#8BA9FC"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  />
</View>
`;

exports[`Standings component renders correctly with no highlighted paramter 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "flex-start",
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingTop": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F2236",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    />
  </View>
  <ActivityIndicator
    animating={true}
    color="#8BA9FC"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  />
</View>
`;
